### PR TITLE
Update meta.yaml

### PR DESCRIPTION
**Pull request by dcfidalgo**
add server dependencies

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dcfidalgo** [comment]
@conda-forge-admin please rerender

**Comment by sugatoray** [comment]
@dcfidalgo Could you please explain why you are adding these dependencies? Are they optional or mandatory minimum requirements for `rubrix`? The way you are adding the dependencies now, that would make them mandatory requirements. If the PyPI package does not use them as mandatory requirements, then this PR needs some changes, before it can be merged.

**Comment by sugatoray** [comment]
The build is failing as the following packages do not exist on conda-forge yet (at least by the names you specified them).

[(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=462363&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=170)

[image](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=462363&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=170)

**Comment by dcfidalgo** [comment]
Hi @sugatoray I added those dependencies because we want to encourage the users to run the Rubrix server themselves instead of using a docker container. The newly added dependencies are necessary for running the server (web app). The pypi package has them as "extra requirements" (that is `pip install rubrix[server]`), so following your advice, what additional changes would be necessary?

> The build is failing as the following packages do not exist on conda-forge yet (at least by the names you specified them).

Yes, I also noticed these missing packages. We can remove the first one, but for the second and third one, I guess I have to create conda recipes since I haven't found a way of specifying dependencies installed via pip.

**Comment by sugatoray** [comment]
@dcfidalgo Yes, the best thing will be to create recipes for each of the missing packages on conda-forge. 

Would you like to start working on creating these recipes? I can guide you on how to do that though.

**Comment by dcfidalgo** [comment]
> Would you like to start working on creating these recipes?

If you like and have some spare time, we could also split the work. Maybe you could create the recipe for `stopwordsiso`, and I will take care of `brotli-asgi`?

**Comment by dcfidalgo** [comment]
I already opened a [PR for brotli-asgi](https://github.com/conda-forge/staged-recipes/pull/18114), let's see when they merge it. @sugatoray If you don't mind, could you have a look at the recipe in case I missed something?

**Comment by dcfidalgo** [comment]
Here goes the PR for the "stopwordsiso" package: https://github.com/conda-forge/staged-recipes/pull/18123

**Comment by sugatoray** [comment]
@dcfidalgo Congratulations on your PRs for those two packages. 🎉👏👏

**Comment by sugatoray** [comment]
@conda-forge-admin, please rerender

**Comment by sugatoray** [comment]
> > Would you like to start working on creating these recipes?
> 
> 
> 
> If you like and have some spare time, we could also split the work. Maybe you could create the recipe for `stopwordsiso`, and I will take care of `brotli-asgi`?

Hi @dcfidalgo , My schedule is going a little tight now. I wasn't able to look into this yesterday. But kudos for getting them done. 

You still need to change the line: `uvicorn[standard]` change to `uvicorn-standard` (https://anaconda.org/conda-forge/uvicorn-standard) as that does not work for conda-forge. Please go to https://conda-forge.org/feedstock-outputs/. You can then search for available conda-forge packages by their names.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -29,11 +29,28 @@ requirements:
     - starlette >=0.13.0,<1.0.0
     - tqdm >=4.27.0
     - wrapt >=1.13.0,<1.14.0
-
+    # server dependencies
+    - fastapi >=0.63.0,<0.64.0
+    - opensearch-py >=1.0.0,<1.1.0
+    - uvicorn[standard] >0.15.0,<0.18.0
```
Comment (dcfidalgo) [inline recipe/meta.yaml]:
```suggestion
    - uvicorn-standard >0.15.0,<0.18.0
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -29,11 +29,28 @@ requirements:
...
+    - smart-open
+    - brotli-asgi >=1.1.0,<1.2.0
+    - Deprecated >=1.2.0,<1.3.0
+    - scikit-learn >=0.24.2
+    - stopwordsiso >=0.6.1,<0.7.0
+    - aiofiles >=0.6.0,<0.7.0
+    - PyYAML >=5.4.1,<5.5.0
+    - python-multipart >=0.0.5,<0.1.0
+    - python-jose[cryptography] >=3.2.0,<3.3.0
```
Comment (dcfidalgo) [inline recipe/meta.yaml]:
```suggestion
    - python-jose >=3.2.0,<3.3.0
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -29,11 +29,28 @@ requirements:
...
+    - passlib[bcrypt] >=1.7.4,<1.8.0
```
Comment (dcfidalgo) [inline recipe/meta.yaml]:
```suggestion
    - passlib >=1.7.4,<1.8.0
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -29,11 +29,28 @@ requirements:
     - starlette >=0.13.0,<1.0.0
     - tqdm >=4.27.0
     - wrapt >=1.13.0,<1.14.0
-
+    # server dependencies
+    - fastapi >=0.63.0,<0.64.0
+    - opensearch-py >=1.0.0,<1.1.0
+    - uvicorn-standard >0.15.0,<0.18.0
+    - smart-open
```
Comment (dcfidalgo) [inline recipe/meta.yaml]:
```suggestion
```

**Comment by sugatoray** [comment]
The build fails during `pip check`. [(error)](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=463402&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=1485)

@dcfidalgo You would need the following three to pass `pip check`. It's better to add pip check for conda-forge recipes, whenever possible. However, if for some reason you cannot get past pip check, the less desirable fix is to omit the pip check section.

- `ecdsa`
- `pyasn1`
- `rsa`

image

```sh
import: 'rubrix.client'
import: 'rubrix.server.server'
import: 'rubrix'
import: 'rubrix.client'
import: 'rubrix.server.server'
+ pip check
python-jose 3.2.0 requires ecdsa, which is not installed.
python-jose 3.2.0 requires pyasn1, which is not installed.
python-jose 3.2.0 requires rsa, which is not installed.
```

**Comment by sugatoray** [comment]
@dcfidalgo I looked at the `setup.cfg` file for rubrix. It appears that the server requirements are available under `extras_require` (which means they are treated as optional dependencies). This would require a different approach: creating multiple subpackages. That will allow people to install either `rubrix-client` or client + `rubrix-server`. 

Let me get back to you on this later.

Opened another PR to update the recipe with subpackages.

- PR: #14

**Comment by dcfidalgo** [comment]
Hi @sugatoray Thank you so much for looking into this. We would strongly prefer if the `conda install rubrix` command would install all dependencies (required + optional), so that the user can run the server via `python -m rubrix`. It would be nice to have another package `conda install rubrix-client` that only installs the client dependencies, but this is really low priority for us.

**Comment by sugatoray** [comment]
Hi @dcfidalgo,

If your PyPI package has `pip install rubrix[server]`, which installs client + server, I would suggest you to follow the same for conda-forge. Currently I am creating the following three sub packages:

- `rubrix` (akin to `pip install rubrix`)
- `rubrix-client` (akin to `pip install rubrix`) -- but the name is more explicit here. 
- `rubrix-server` (akin to `pip install rubrix[server]`)

The client is a subset of the `client + server` setup. This is why, a base package is first created with client dependencies and then that base package is used for the server sub package. 

The naming of these packages is open for discussion. There is a build error that is currently blocking progress. So, please let me know your thoughts.

**Comment by dcfidalgo** [comment]
> 
    rubrix (akin to pip install rubrix)
    rubrix-client (akin to pip install rubrix) -- but the name is more explicit here.
    rubrix-server (akin to pip install rubrix[server])

Ok, sounds reasonable, not sure though if the `rubrix-client` sub-package is really needed, maybe we can omit it for now. I guess it's better to start with too few sub-packages than with too many. 

I pushed a final commit to this PR, where I install the missing dependencies to make `pip check` work, in case you want to do the same in the sub-package.

Let's leave this PR open for now, and see how your other PR evolves.

**Comment by sugatoray** [comment]
> Ok, sounds reasonable, not sure though if the rubrix-client sub-package is really needed, maybe we can omit it for now. I guess it's better to start with too few sub-packages than with too many.

@dcfidalgo Yes, that makes sense. I switched off making `rubrix-client` for now. 

| PyPI | Conda | Package Type |
|:---|:---|:---|
| `pip install rubrix` | `conda install -c conda-forge rubrix` | Client |
| `pip install rubrix[server]` | `conda install -c conda-forge rubrix-server` | Client + Server |

> I pushed a final commit to this PR, where I install the missing dependencies to make pip check work, in case you want to do the same in the sub-package.

Thank you, for getting this sorted. I added these and the recipe now runs without any error. :tada:

**Comment by sugatoray** [comment]
The other PR is ready for merger. :tada:

- https://github.com/conda-forge/rubrix-feedstock/pull/14#issuecomment-1048361404

**Comment by dcfidalgo** [comment]
Closing this due to #14

**Pull request closed without merging**